File: ios/Tradeguruelectrical/Models/UserTier.swift
import Foundation
import SwiftUI

enum UserTier: String, Codable, Sendable {
    case free
    case pro
    case unlimited

    var displayName: String {
        switch self {
        case .free: "Free"
        case .pro: "Pro"
        case .unlimited: "Unlimited"
        }
    }

    var color: Color {
        switch self {
        case .free: .tradeTextSecondary
        case .pro: .modeLearn
        case .unlimited: .tradeGreen
        }
    }
}

File: android/app/src/main/java/com/tradeguru/electrical/models/UserTier.kt
package com.tradeguru.electrical.models

enum class UserTier(val value: String, val displayName: String) {
    FREE("free", "Free"),
    PRO("pro", "Pro"),
    UNLIMITED("unlimited", "Unlimited");

    companion object {
        fun fromValue(value: String): UserTier? =
            entries.find { it.value == value }
    }
}
